pkg/realizer/pipeline/realizer.go

```diff
@@ -109,3 +122,24 @@ func (p *pipelineRealizer) Realize(ctx context.Context, pipeline *v1alpha1.Pipel
 
 	return RunTemplateReadyCondition(), outputs, stampedObject
 }
+
+func resolveSelector(selector *v1alpha1.ResourceSelector, repository repository.Repository) (*map[string]interface{}, error) {
+	if selector == nil {
+		return nil, nil
+	}
+	queryObj := &unstructured.Unstructured{}
+	queryObj.SetGroupVersionKind(schema.FromAPIVersionAndKind(selector.Resource.APIVersion, selector.Resource.Kind))
+	queryObj.SetLabels(selector.MatchingLabels)
+
+	results, err := repository.ListUnstructured(queryObj)
+	if err != nil {
+		return nil, fmt.Errorf("could not list objects matching selector: %w", err)
+	}
+	if len(results) == 1 {
```

Comment:
nit: I think we usually do early returns for failure cases - might be worth sticking to that